src/runtime/bddStepInvoker.ts

```diff
@@ -46,16 +47,22 @@ export class BddStepInvoker {
 
     const fixturesArg = this.getStepFixtures(stepFixtures || {});
 
-    await runStepWithLocation(this.bddContext.test, stepTextWithKeyword, location, () => {
+    await runStepWithLocation(this.bddContext.test, stepTextWithKeyword, location, async () => {
       // Although pw-style does not expect usage of world / this in steps,
       // some projects request it for better migration process from cucumber.
       // Here, for pw-style we pass empty object as world.
       // See: https://github.com/vitalets/playwright-bdd/issues/208
-      return matchedDefinition.definition.fn.call(
+      const beforeHooksToRun = getStepHooksToRun('before', this.bddContext.tags);
+      await runStepHooks(beforeHooksToRun, { ...fixturesArg, $bddContext: this.bddContext });
+      const use = await matchedDefinition.definition.fn.call(
```

Comment:
Let's name it `result` to not confuse with Playwright's `use()` function.

Reply:
changed the variable name to `result`